### PR TITLE
AirflowNetwork Diff-Free Refactor

### PR DESCRIPTION
**Pull request by jasondegraw**
Pull request overview
---------------------
Refactor and reorganize AirflowNetwork to make performance improvements easier to implement. The main files are now in a separate subdirectory and the code is partially disentangled from EnergyPlus proper. Additional disentanglement will likely trigger diffs. This pull request is largely diff-free (the only diffs are due to some reordering of elements in memory that are a side effect of using the new input processor).

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things

### CODE REVIEW

**Comment by jasondegraw** [comment]
@Myoldmopar @lgu1234 At this point, there are still some reorganization-type things to do, but I keep triggering diffs. So I'm going to stop here for now, diffs can wait until the next release. I'm tagging you both as reviewers, but feel free to disagree.

File: scripts/dev/license-check.py
Comment (Myoldmopar) [inline scripts/dev/license-check.py:32]: So are we just not checking on the subdirectories any more?  Maybe it's just a little hard to tell the intent from this small diff.

File: src/EnergyPlus/AirflowNetwork/CMakeLists.txt
Comment (Myoldmopar) [inline src/EnergyPlus/AirflowNetwork/CMakeLists.txt:23]: Once all the comments are cleaned out of here, this looks good to go.  I like that it is a little standalone static lib now.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp
Comment (Myoldmopar) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp:361]: At some point, this base class will presumably grow to include (at least) a pure virtual `calculate()` method, right?

File: src/EnergyPlus/AirflowNetwork/src/Elements.cpp
Comment (Myoldmopar) [inline src/EnergyPlus/AirflowNetwork/src/Elements.cpp:242]: Would it be beneficial for these two arrays to be const refs?

File: src/EnergyPlus/AirflowNetwork/src/Elements.cpp
Comment (Myoldmopar) [inline src/EnergyPlus/AirflowNetwork/src/Elements.cpp]: Feel free to clean out these comments as a part of your diff-free cleanup.

File: src/EnergyPlus/CMakeLists.txt
Comment (Myoldmopar) [inline src/EnergyPlus/CMakeLists.txt:646]: Maybe a couple more comments to trim before merging, unless you want to leave them in as clues for later refactoring.

File: src/EnergyPlus/InputProcessing/InputProcessor.cc
Comment (Myoldmopar) [inline src/EnergyPlus/InputProcessing/InputProcessor.cc:222]: :+1: :+1: :+1: This will be very handy for other work that relies on raw JSON input processing.  @mbadams5 any thoughts or concerns here?

File: src/EnergyPlus/InputProcessing/InputProcessor.cc
Comment (Myoldmopar) [inline src/EnergyPlus/InputProcessing/InputProcessor.cc]: Bleck, something got funny with the indentation here.  I don't like the data word just hanging out down there.

File: src/EnergyPlus/ZoneEquipmentManager.cc
Comment (Myoldmopar) [inline src/EnergyPlus/ZoneEquipmentManager.cc:4732]: I'm inclined to ask if you'd wrap this in braces while you're making the rest of your changes.

File: src/EnergyPlus/ZoneEquipmentManager.cc
Comment (Myoldmopar) [inline src/EnergyPlus/ZoneEquipmentManager.cc:5548]: Same here about braces.

File: tst/EnergyPlus/unit/RoomAirflowNetwork.unit.cc
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/RoomAirflowNetwork.unit.cc:218]: Why do these square braces make me so happy?

**Review by Myoldmopar — APPROVED** [review]
See my comments for just a few things.  Technically I'm requesting a few changes, but they are almost all just cleaning up commented stuff.  So I'm marking it approved.  This is a really great prefactoring step with no diffs.  The two CI warnings are super acceptable.  I will plan on building this locally once more right before merge to ensure everything is good when merging in develop, but otherwise this should be able to drop in pretty easy.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp
Comment (lgu1234) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp:1328]: Currently, we use prefix, Multizone and DisSys, to represent which component is used in multizone airflow or distribution airflow calculations. By removing prefix, it is not easy to recognize component usage. Is it possible to use some flag to show component usage?

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Properties.hpp
Comment (lgu1234) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Properties.hpp:101]: Although these air properties are used in the AirflowNetwork right now, the properties may be used in other modules in the future. Is it possible to make a global data?

File: src/EnergyPlus/AirflowNetwork/src/Elements.cpp
Comment (lgu1234) [inline src/EnergyPlus/AirflowNetwork/src/Elements.cpp:387]: Is it essential to have a return value for each calculate function?

File: src/EnergyPlus/AirflowNetwork/src/Properties.cpp
Comment (lgu1234) [inline src/EnergyPlus/AirflowNetwork/src/Properties.cpp]: I know this is not your change. I prefer to revise a warning message slightly to add either lower or Upper limit with number.

ShowWarningMessage("Air temperature out of lower limit of -20C for conductivity calculation");

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (lgu1234) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:270]: DO you plan to remove PZ(i) = AirflowNetworkNodeSimu(i).PZ; and use properties[i].pressure = AirflowNetworkNodeSimu(i).PZ;?

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (lgu1234) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:313]: Do we have a plan to make array index starting from 0, instead of 1?

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (lgu1234) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:376]: Do we have any benefit to redefine "int i=1..." for every for loop?

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (lgu1234) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp]: These line may be deleted, because never used.

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (lgu1234) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:3342]: We have an air density function in Psy module. Do we need an additional AIRDENSITY function?

File: scripts/dev/license-check.py
Comment (jasondegraw) [reply scripts/dev/license-check.py:32]: No, I've changed to using Python's glob module differently so it should now hit all subdirectories now.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp:361]: Yes, it didn't quite make into this PR.

File: src/EnergyPlus/AirflowNetwork/src/Elements.cpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/src/Elements.cpp:242]: No, those are two of the three returns out of these functions. I thought about wrapping all three into a struct, but I'm not quite sold on that yet.

File: src/EnergyPlus/InputProcessing/InputProcessor.cc
Comment (jasondegraw) [reply src/EnergyPlus/InputProcessing/InputProcessor.cc]: Blame the tyranny of clang-format for that.

File: src/EnergyPlus/ZoneEquipmentManager.cc
Comment (jasondegraw) [reply src/EnergyPlus/ZoneEquipmentManager.cc:4732]: I can do that.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp:1328]: I'm not a huge fan of labeling like that in the naming (the object names get way too long), but I'll try to think of something that'll help without bloating the names.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Properties.hpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Properties.hpp:101]: Yes, that can be done.

File: src/EnergyPlus/AirflowNetwork/src/Elements.cpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/src/Elements.cpp:387]: Yes, the intent here is to move to using polymorphism, so they all need to have the same signature.

File: src/EnergyPlus/AirflowNetwork/src/Properties.cpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/src/Properties.cpp]: I can make that change.

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/src/Solver.cpp:270]: Yes, at some point. It's a change that goes with other changes that end up causing diffs, so that'll come in a later PR.

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/src/Solver.cpp:313]: Yes, removal of all the Objexx arrays is part of the refactoring plan.

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/src/Solver.cpp:376]: Generally speaking I'd expect the compiler to do the best thing, so there may not be much performance benefit but I wouldn't expect a performance hit either. However, this way the scope where `i` is valid is just in the loop, which protects against unintentional use and clashes.

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/src/Solver.cpp:3342]: That's actually a `#define` to `PsyRhoAirFnPbTdbW`, so it is the same thing.

**Review by lgu1234 — APPROVED** [review]

File: src/EnergyPlus/AirflowNetwork/src/Elements.cpp
Comment (Myoldmopar) [reply src/EnergyPlus/AirflowNetwork/src/Elements.cpp:242]: :+1:

**Comment by Myoldmopar** [comment]
@jasondegraw the license-check script is failing.  Running locally with Python 2, which is what the CI machines have, I get:

```
TypeError: glob() got an unexpected keyword argument 'recursive'
```

Looks like the `recursive` keyword was added to Glob in Python 3.5.  
I would suggest changing the code to grab all the files recursively in a Py2/3 way like:

```
files = []
for root, _, filenames in os.walk(source_dir):
    for filename in fnmatch.filter(filenames, '*.cc'):
        files.append(os.path.join(root, filename))
```

**Comment by jasondegraw** [comment]
@Myoldmopar Sigh. One more year.

**Comment by Myoldmopar** [comment]
https://pythonclock.org/

**Comment by Myoldmopar** [comment]
@jasondegraw I fixed the recursive call and verified it works by botching a source file deep in a source subdirectory.  It properly warned.  I'm merging this.  Thanks for this refactor!

**Pull request merged**
